lib/components/helpers/pretty-select-value.js

```diff
@@ -258,11 +258,9 @@ export default createReactClass({
 
     if (this.state.isEnteringCustomValue || (!isDefaultValue && !currentChoice && currentValue)) {
       if (this.hasCustomField()) {
-        const {choices} = this.props;
+        const {choices, config} = this.props;
         const customChoice = _.find(choices, (choice) => choice.action === 'enter-custom-value');
-        if (customChoice && customChoice.label) {
-          return customChoice.label;
-        }
+        return config.customChoiceDisplayValue(customChoice, currentValue) || currentValue;
```

Comment:
The only thing I can think of to improve this is potentially refactoring the `|| currentValue`, as it duplicates the `return currentValue` below.